src/test/java/com/team1/dealerApp/user/UserServiceTest.java

```diff
@@ -0,0 +1,177 @@
+        Movie whatchedMovie = new Movie("Il signore degli anelli: il ritorno del Re", Genre.FANTASY, castMovie, "Peter Jackson", Year.of(2003), 30.00, 10.00, "Il ritorno del Re", 4.0f, VideoStatus.RENTABLE, 110);
+        Movie whatchedMovie2 = new Movie("Alice nel paese delle Meraviglie", Genre.ANIMATION, castMovie2, "Clyde Geronimi", Year.of(1951), 30.00, 10.00, "Alice nel paese delle meraviglie", 2.0f, VideoStatus.RENTABLE, 110);
+        List<Movie> whatchedMovieList = new ArrayList<>();
+        whatchedMovieList.add(whatchedMovie);
+        whatchedMovieList.add(whatchedMovie2);
+
+        List<String> castShow = Arrays.asList("Bryan Cranston", "Aaron Paul", "Giancarlo Esposito");
+        List<String> castShow2 = Arrays.asList("Antony Starr", "Karl Urban", "Jack Quaid");
+        TvShow whatchedTvShow = new TvShow("Breaking bad", Genre.DRAMA, castShow, "Vince Gilligan", Year.of(2006), 50.0, 10.0, "Un prof si ammala e inizia a fare la droga", 5.0f, VideoStatus.RENTABLE, 6, 52);
+        TvShow whatchedTvShow2 = new TvShow("The Boys", Genre.ACTION, castShow2, "Erik Kripke", Year.of(2019), 60.0, 10.0, "Patriota impazzisce", 4.6f, VideoStatus.RENTABLE, 5, 40);
+        List<TvShow> whatchedTvShowList = new ArrayList<>();
+        whatchedTvShowList.add(whatchedTvShow);
+        whatchedTvShowList.add(whatchedTvShow2);
+
+
+        user = new User();
+
+        userId = UUID.randomUUID();
+
+        user.setId(userId);
+        user.setFirstName("Mario");
+        user.setLastName("Rossi");
+        user.setEmail("mario.rossi@gmail.com");
+        user.setPhoneNumber("3331234567");
+        user.setSubscriptionStatus(SubscriptionStatus.FULL_SUBSCRIPTION);
+        user.setWatchedMovies(whatchedMovieList);
+        user.setWatchedShows(whatchedTvShowList);
+
+
+        createUserDTO = new CreateUserDTO();
+
+        createUserDTO.setFirstName("Mario");
+        createUserDTO.setLastName("Rossi");
+        createUserDTO.setEmail("mario.rossi@gmail.com");
+        createUserDTO.setPhoneNumber("3331234567");
+        createUserDTO.setSubscriptionStatus(SubscriptionStatus.FULL_SUBSCRIPTION);
+        createUserDTO.setWatchedMovies(whatchedMovieList);
+        createUserDTO.setWatchedShows(whatchedTvShowList);
+
+
+        rentalId = 1L;
+
+        rental = new Rental();
+        rental.setId(rentalId);
+        rental.setStartDate(LocalDateTime.now());
+        rental.setEndDate(LocalDateTime.now().plusDays(14));
+        rental.setRentalStatus(RentalStatus.ACTIVE);
+        rental.setMovies(whatchedMovieList);
+        rental.setTvShows(whatchedTvShowList);
+        rental.setRentalPrice(20.00);
+
+        List<Rental> rentalList = new ArrayList<>();
+        rentalList.add(rental);
+
+        user.setRentals(rentalList);
+
+    }
+
+
+    //Test per getUserDTOById if-> NoSuchElementException
+    @Test
+    void testGetUserDTOById_NoSuchElementException() {
+        when(userRepository.findById(userId)).thenReturn(Optional.empty());
```

Comment:
Qui ad esempio non utilizzi nessuno degli oggetti che ha inizializzato, ma mettendoli nella fase di setup vengono creati comunque ogni volta.

La fase di setup va tenuta più snella possibile e vanno inserite in essa solo oggetti e mock che vengono utilizzati da un numero considerevole di unit test.